contributors/design-proposals/auth/pod-restriction.md

```diff
@@ -0,0 +1,477 @@
+bind to the target objects service account. ServiceAccount usage isn't actually authorized
+separately from namespace usage, so this just ends up being namespace-level authorization. However,
+because PodSecurityPolicy is authorized by either the source or target user, this ends up being less
+secure, since the Pod can use it's service account to create pods in other namespaces with the same
+access level. For instance, using PodSecurityPolicy you cannot have a controller that runs as
+privileged without opening up all pods it creates to running as privileged.
+
+**Namespaced Objects**: This approach has its advantages, but it doesn't scale well to large
+organizations with many namespaces (without a system on top managing all the policies across
+namespaces). See extensions below for how we might use namespaced objects with PodRestrictions.
+
+**Bind to Pod labels**: Pod labels don't have any authorization of usage without something like
+PodRestriction, so binding to labels isn't sufficient no it's own.
+
+**Bind to Namespaces**: The approach I'm recommending. Cluster level objects use a namespace
+selector to select the namespaces it applies to. It is consistent with the authorization of create
+requests, which are authorized at the namespace level. A small number of policies can be managed at
+the cluster level, and mapped to the namespaces they should apply to.
+
+#### Optional Extensions
+
+-   Namespaced PodRestrictions - PodRestriction objects can be created in a namespace, so that
+    namespace admins can place additional restrictions within their namespace.
+
+-   Additional Bindings - The NamespaceSelector is not a top-level field so that additional binding
+    approaches may be added in the future. Examples may include: ServiceAccounts, LabelSelector,
+    NodeSelector, etc.
+
+### Policy Matching - Union or Intersection?
+
+**Intersection** refers to the model where a Pod must be valid under all PodRestrictions that
+pertain to it. The advantage of intersecting is that restrictions can be composed. For example, I
+could have a policy that frontend & backend Pods cannot share a node, and a separate policy that all
+production Pods must drop privileges. Without the ability to intersect those policies, I need 3
+different versions (either, or, both). A problem with this approach is that running a privileged
+namespace (e.g. kube-system) means that every restriction needs an exception in the
+NamespaceSelector.
+
+**Union** refers to the model where a Pod must be valid under any PodRestriction that is bound to
+it. This approach makes it easier to reason about what the restrictions on a given namespace are,
+and easier to add exceptions for namespaces, but can lead to a combinatorial explosion of different
+policies that can become unmanageable.
+
+**Proposal**: Why not both? The default behavior is intersection, but you can punch a hole through
```

Comment:
@tallclair - what if a Policy specifies intersection and another one union, and both are applicable, how do we resolve this ?

Reply:
I was proposing that the "union" (accept) policies are checked first. Alternatively, the "firewall" approach I mentioned uses priority and favors deny.